subsys/bluetooth/mesh/net.c

```diff
@@ -266,14 +266,26 @@ bool bt_mesh_net_iv_update(uint32_t iv_index, bool iv_update)
 		}
 
 		if (iv_index == bt_mesh.iv_index + 1 && !iv_update) {
-			BT_WARN("Ignoring new index in normal mode");
-			return false;
-		}
+			if (!(IS_ENABLED(CONFIG_BT_MESH_IV_UPDATE_TEST) &&
+			      atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_TEST))) {
+				if (bt_mesh.ivu_duration < BT_MESH_IVU_MIN_HOURS) {
+					BT_WARN("Ignoring new index in normal "
+						"mode before minimum duration");
+					return false;
+				}
+			}
 
-		if (!iv_update) {
-			/* Nothing to do */
-			BT_DBG("Already in Normal state");
-			return false;
+			/* If the detected IV Index is current IV Index + 1, and
+			 * node is in Normal mode of operation, it means
+			 * previous IV update has been missed, immediately
+			 * update the IV Index without going to IV update in
+			 * progress state.
+			 */
+			BT_WARN("Performing IV Index Recovery");
+			bt_mesh_rpl_clear();
+			bt_mesh.iv_index = iv_index;
+			bt_mesh.seq = 0U;
+			goto do_update;
```

Comment:
Can this be combined with lines 260-266, perhaps? e.g. change line 260 to `if (iv_index > bt_mesh.iv_index)`, then insert the guard on lines 268-276 on line 261